packages/protocolbeat/src/panel-terminal/TerminalPanel.tsx

```diff
@@ -0,0 +1,102 @@
+import { useQuery } from '@tanstack/react-query'
+import { useEffect, useRef, useState } from 'react'
+import { useParams } from 'react-router-dom'
+import { executeCommand, getProject } from '../api/api'
+import { useTerminalStore } from '../store/store'
+
+export function TerminalPanel() {
+  const { project } = useParams()
+  const [discoveryChain, setDiscoveryChain] = useState<string | undefined>(
+    undefined,
+  )
+  const abortControllerRef = useRef<AbortController>()
+  const outputRef = useRef<HTMLDivElement>(null)
+  const { output, isRunning, addOutput, setIsRunning, clear } =
+    useTerminalStore()
+
+  if (!project) {
+    throw new Error('Cannot use component outside of project page!')
+  }
+
+  const getProjectResponse = useQuery({
+    queryKey: ['projects', project],
+    queryFn: () => getProject(project),
+  })
+  const chains = getProjectResponse.data?.chains ?? []
+
+  useEffect(() => {
+    clear()
+  }, [])
+
+  useEffect(() => {
+    return () => {
+      // Cleanup on unmount
+      abortControllerRef.current?.abort()
+    }
+  }, [])
+
+  useEffect(() => {
+    if (outputRef.current) {
+      outputRef.current.scrollTop = outputRef.current.scrollHeight
```

Comment:
This is always frustrating if implemented in this way. You only want to scroll if it was already fully scrolled. Otherwise you want to keep where the user scrolled to.

Reply:
This is not fixed as I find it a bit too challenging for now (my solutions never work reliably). I'll do it as a separate PR when I have more time or pass it to someone better with frontend code.